File: python/day-03/part1.py
with open("input.txt", "r") as reader:
    data = reader.readlines()

data = """467..114..
...*......
..35..633.
......#...
617*......
.....+.58.
..592.....
......755.
...$.*....
.664.598.."""
data = data.split("\n")

sum = 0
for h, i in enumerate(data):
    for k in range(len(i)):
        if i[k].isnumeric():
            valid = True
            value = 0;
            for m in range(k, len(i)):
                if not i[m].isnumeric():
                    value = int(i[k:m])
                    if not (data[h][m] == "."):
                        valid = False
                    if m-1 >= 0 and data[h][m-1] != ".":
                        valid = False
                    if k-1 >= 0 and h-1 >= 0 and data[h-1][m] != ".":
                        valid = False
                    if k-1 >= 0 and h-1 >= 0 and data[h-1][m-1] != ".":
                        valid = False
                    if h+1 <= len(i)-1 and  data[h+1][m] != ".":
                        valid = False
                    if h+1 <= len(i)-1 and data[h+1][m-1] != ".":
                        valid = False
                    for n in range(m-k):
                        if h-1 
                        if (data[h-1][k+n] == "." and
                            data[h+1][k+n] == "."):
                            valid = False
                            break
            if valid:
                sum += value
print(sum)

File: rust/day-03/src/bin/part1.rs
use std::fs;

fn main() {
    let file_path = "input1.txt";
    let input = fs::read_to_string(file_path).unwrap();
    println!("{}", part1(&input));
}

fn part1(input: &str) -> String {
    let data = input.split("\n").collect::<Vec<&str>>();
    for line in data {}
    todo!()
}

#[cfg(test)]
mod test {
    #[test]
    fn given_io() {
        use super::*;

        let input = "";
        let result = part1(input);
        assert_eq!("", result)
    }
}
